sdks/java/harness/src/test/java/org/apache/beam/fn/harness/debug/OutputSamplerTest.java

```diff
@@ -80,14 +84,50 @@ public void testActsLikeCircularBuffer() throws Exception {
     // The first 10 are always sampled, but with maxSamples = 5, the first ten are downsampled to
     // 4..9 inclusive. Then,
     // the 20th element is sampled (19) and every 20 after.
-    List<byte[]> expected = new ArrayList<>();
+    List<BeamFnApi.SampledElement> expected = new ArrayList<>();
     expected.add(encodeInt(19));
     expected.add(encodeInt(39));
     expected.add(encodeInt(59));
     expected.add(encodeInt(79));
     expected.add(encodeInt(99));
 
-    List<byte[]> samples = outputSampler.samples();
+    List<BeamFnApi.SampledElement> samples = outputSampler.samples();
     assertThat(samples, containsInAnyOrder(expected.toArray()));
   }
+
+  /**
+   * Test that sampling a PCollection while retrieving samples from multiple threads is ok.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testConcurrentSamples() throws Exception {
+    VarIntCoder coder = VarIntCoder.of();
+    OutputSampler<Integer> outputSampler = new OutputSampler<>(coder, 100000, 1);
```

Comment:
```suggestion
    OutputSampler<Integer> outputSampler = new OutputSampler<>(coder, 100000, 1);
```

You can use a small maxElements size like 10. Using a bigger number may decrease contention. Also using sampleEveryN of 2 means that we swap between choosing to sample and not to sample often instead of sampling every element.

Reply:
Done